cypress/support/utils/product.js

```diff
@@ -61,79 +64,110 @@ export class Product {
 
-    cy.writeIntoStringField('Description', product.description);
+      cy.writeIntoStringField('Description', product.description);
 
     cy.getFieldValue('IsStocked').then(isIsStockedValue => {
-      if (product.isStocked && !isIsStockedValue) {
-        cy.clickOnCheckBox('IsStocked');
-      }
-    });
+        if (product.isStocked && !isIsStockedValue) {
+          cy.clickOnCheckBox('IsStocked');
+        }
+      });
     cy.getFieldValue('IsPurchased').then(isPurchasedValue => {
-      if (product.isPurchased && !isPurchasedValue) {
-        cy.clickOnCheckBox('IsPurchased');
-      }
-    });
+        if (product.isPurchased && !isPurchasedValue) {
+          cy.clickOnCheckBox('IsPurchased');
+        }
+      });
     cy.getFieldValue('IsSold').then(isSoldValue => {
-      if (product.isSold && !isSoldValue) {
-        cy.clickOnCheckBox('IsSold');
-      }
-    });
+        if (product.isSold && !isSoldValue) {
+          cy.clickOnCheckBox('IsSold');
+        }
+      });
     cy.getFieldValue('IsDiverse').then(isDiverseValue => {
-      if (product.isDiverse && !isDiverseValue) {
-        cy.clickOnCheckBox('IsDiverse');
-      }
-    });
+        if (product.isDiverse && !isDiverseValue) {
+          cy.clickOnCheckBox('IsDiverse');
+        }
+      });
 
-    cy.getFieldValue('ProductType').then(productTypeValue => {
-      const productType = getLanguageSpecific(product, 'productType');
+      cy.getFieldValue('ProductType').then(productTypeValue => {
+        const productType = getLanguageSpecific(product, 'productType');
 
-      if (productType != productTypeValue) {
-        cy.selectInListField('ProductType', productType);
-      }
-    });
+        if (productType != productTypeValue) {
+          cy.selectInListField('ProductType', productType);
+        }
+      });
 
-    cy.getFieldValue('C_UOM_ID').then(uomValue => {
-      const c_uom = getLanguageSpecific(product, 'c_uom');
+      cy.getFieldValue('C_UOM_ID').then(uomValue => {
+        const c_uom = getLanguageSpecific(product, 'c_uom');
 
-      if (c_uom && c_uom != uomValue) {
-        cy.selectInListField('C_UOM_ID', c_uom);
+        if (c_uom && c_uom != uomValue) {
+          cy.selectInListField('C_UOM_ID', c_uom);
+        }
+      });
+
+      if (product.productPrices.length > 0) {
+        product.productPrices.forEach(function (price) {
+          Product.applyProductPrice(price);
+        });
+        cy.get('table tbody tr').should('have.length', product.productPrices.length);
       }
     });
+  }
 
-    if (product.prices.length > 0) {
-      product.prices.forEach(function(price) {
-        applyProductPrice(price);
-      });
-      cy.get('table tbody tr').should('have.length', product.prices.length);
-    }
-  });
+  /**
+   * See complaint at ProductPrice class.
+   */
+  static applyProductPrice(productPrice) {
+    // const m_pricelist_version = getLanguageSpecific(productPrice, 'm_pricelist_version');
```

Comment:
pls use the getLanguageSpecific so we can run this with different languages

Reply:
No need for this trl since product price is its own data type with its own name.